test/unit/auth.js

```diff
@@ -113,4 +115,224 @@ describe('DropboxAuth', () => {
       }
     });
   });
+
+  describe('clientSecret', () => {
+    it('can be set in the constructor', () => {
+      const dbx = new Dropbox({ clientSecret: 'foo' });
+      chai.assert.equal(dbx.auth.getClientSecret(), 'foo');
+    });
+
+    it('is undefined if not set in constructor', () => {
+      const dbx = new Dropbox();
+      chai.assert.equal(dbx.auth.getClientSecret(), undefined);
+    });
+
+    it('can be set after being instantiated', () => {
+      const dbx = new Dropbox();
+      dbx.auth.setClientSecret('foo');
+      chai.assert.equal(dbx.auth.getClientSecret(), 'foo');
+    });
+  });
+
+  describe('refreshToken', () => {
+    it('can be set in the constructor', () => {
+      const dbxAuth = new DropboxAuth({ refreshToken: 'foo' });
+      chai.assert.equal(dbxAuth.getRefreshToken(), 'foo');
+    });
+
+    it('is undefined if not set in constructor', () => {
+      const dbxAuth = new DropboxAuth();
+      chai.assert.equal(dbxAuth.getRefreshToken(), undefined);
+    });
+
+    it('can be set after being instantiated', () => {
+      const dbxAuth = new DropboxAuth();
+      dbxAuth.setRefreshToken('foo');
+      chai.assert.equal(dbxAuth.getRefreshToken(), 'foo');
+    });
+  });
+
+  describe('AccessTokenExpiresAt', () => {
+    it('can be set in the constructor', () => {
+      const date = new Date();
+      const dbxAuth = new DropboxAuth({ accessTokenExpiresAt: date });
+      chai.assert.equal(dbxAuth.getAccessTokenExpiresAt(), date);
+    });
+
+    it('is undefined if not set in constructor', () => {
+      const dbxAuth = new DropboxAuth();
+      chai.assert.equal(dbxAuth.getAccessTokenExpiresAt(), undefined);
+    });
+
+    it('can be set after being instantiated', () => {
+      const dbxAuth = new DropboxAuth();
+      const date = new Date();
+      dbxAuth.setAccessTokenExpiresAt(date);
+      chai.assert.equal(dbxAuth.getAccessTokenExpiresAt(), date);
+    });
+  });
+
+  describe('generatePKCECodes', () => {
```

Comment:
Note for future reference. Test that when you use PKCE flow, in getAuthUrl, that it calls this function